sphinx/application.py

```diff
@@ -151,12 +151,9 @@ def __init__(self, srcdir: str, confdir: Optional[str], outdir: str, doctreedir:
         self.srcdir = abspath(srcdir)
         self.outdir = abspath(outdir)
         self.doctreedir = abspath(doctreedir)
-        self.confdir = confdir
+        self.confdir = abspath(confdir) if confdir else confdir
         if self.confdir:  # confdir is optional
-            self.confdir = abspath(self.confdir)
-            if not path.isfile(path.join(self.confdir, 'conf.py')):
-                raise ApplicationError(__("config directory doesn't contain a "
-                                          "conf.py file (%s)") % confdir)
+            Config.get_config_file(self.confdir)
```

Comment:
is it necessary to have this check here, or as you say below, is it's enough to just raise an error on reading instead of checking existence?

Reply:
Good point. It seems nobody touches the config file until L214. So it's enough to raise an exception inside `Config.read()`.

Reply:
done 👍